### PR TITLE
feat: init token routing

### PR DESCRIPTION
**Pull request by hetfly**

### CODE REVIEW

**Comment by hetfly** [comment]
@WeiLi512 
1) token.totalSupply from useAllTokensQuery() shows circulating supply instead of total supply, at least for ICX
2) I need to get info about tokens 24h volume, can you point me to some docs where I can find out how ?
thanks

**Comment by hetfly** [comment]
@parrot9design  @WeiLi512  please review,
we are missing ICX total supply and 24h volume for all tokens,
I assume total supply is the same as circulating supply for BALN, sICX a bnUSD.
Do we ship as is or do we wait ?

**Review by parrot9design — CHANGES_REQUESTED** [review]
>we are missing ICX total supply and 24h volume for all tokens,
I assume total supply is the same as circulating supply for BALN, sICX a bnUSD.

Yep, total and circulating supply will be the same for sICX, BALN, and bnUSD. The ICX total supply and 24h volume for all coins may take a bit more work to pull through, so for now, let's omit the third line of data and change `Total supply` -> `Circulating supply`:

Screen Shot 2021-07-22 at 2 42 17 PM

---

In the ICX summary, there's a space where we can show a dynamic ICX reward rate. This figure is already used in the Collateral section on the Stats page:

>  If you stake ICX in a supported wallet, you'll earn regular staking rewards, based on ICON's variable reward rate (currently x% per year). 

---

Let's also get the arrows in place for each of the tokens, to make it clear you can interact with them:

**Current** | **Design**
----------- | ------------
Screen Shot 2021-07-22 at 2 31 28 PM |Screen Shot 2021-07-22 at 2 31 23 PM

---

The summary text is also a bit too large. Let's give it these styles:

```
font-size: 18px;
line-height: 32px;
```

**Comment by hetfly** [comment]
@parrot9design do you see the links without arrows here as well ? If so, please specify your browser version
https://balanced-network-info-i07pt108g-balanced-dao.vercel.app/

**Comment by hetfly** [comment]
Updated based on the feedback, but the current staking rewards value in collateral section is a static one. 
I didn't find a way to get it dynamically.

**Comment by parrot9design** [comment]
> @parrot9design do you see the links without arrows here as well ? If so, please specify your browser version

I see the arrows now. They make the token names harder to read at smaller screen sizes, though – can we increase the width of the asset column to avoid splitting words in half?

**Current** | ** Preferred**
----------- | --------------
Screen Shot 2021-07-23 at 4 12 15 PM | Screen Shot 2021-07-23 at 4 13 47 PM

---

> the current staking rewards value in collateral section is a static one.
I didn't find a way to get it dynamically.

That I wasn't aware of, thanks for bringing it to my attention! It looks like you've found a way to get it dynamically now? Where are you pulling the data from?

**Comment by hetfly** [comment]
Token name width is fixed,
staking rewards are still static, I just updated the current value.

**Comment by parrot9design** [comment]
This is great, thanks @hetfly. I've opened an issue to get the ICX staking apy resolved: https://github.com/balancednetwork/balanced-network-info/issues/29

Ideally I would like to get that implemented before we publish this, to minimise the number of static data points we need to maintain.

File: src/pages/TokenPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+import styled from 'styled-components';
+
+import Footer from 'components/Footer';
+import Header from 'components/Header';
+import { CurrencyKey } from 'constants/currency';
+import { tokenInfo } from 'constants/tokenInfo';
+import { getCurrencyKeyIcon } from 'utils';
+import { getFormattedNumber } from 'utils/formatter';
+
+const Container = styled(Box)`
```
Comment (WeiLi512) [inline src/pages/TokenPage/index.tsx:16]: I can see some same components(`Container`, `CurrencyIcon`,..) used in the codebase. Please DRY the code as you can do.

**Review by WeiLi512 — CHANGES_REQUESTED** [review]

**Comment by parrot9design** [comment]
@hetfly it looks like we have a dynamic APY for ICX staking now: https://github.com/balancednetwork/balanced-network-info/pull/34

File: src/components/TokenDetails/ICXDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        <CurrencyIcon currencyKey={token.symbol} width={75} height={75} />
+        <Text>{token.name}</Text>
+      </TokenName>
+      <TokenDetails>
+        <TokenInfo>
+          {`ICON is a decentralized blockchain network focused on interoperability, and ICX is the native cryptocurrency. If you stake ICX in a supported wallet, you'll earn regular staking rewards, based on ICON's variable reward rate (currently ${
+            collateralInfo.stakingAPY ? getFormattedNumber(collateralInfo.stakingAPY, 'percent2') : '-'
+          } per year).`}
+          <LineBreak />
+          {`On Balanced, you can use ICX as collateral to borrow synthetic assets, like Balanced Dollars.`}
```
Comment (parrot9design) [inline src/components/TokenDetails/ICXDetail.tsx:90]:
```suggestion
          {`On Balanced, you can use ICX as collateral to borrow synthetic assets, like Balanced Dollars. You'll also need to keep at least 1 ICX in your wallet at all times to cover transaction fees.`}
```

File: src/components/TokenDetails/SICXDetail.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <>
+      <TokenName>
+        <CurrencyIcon currencyKey={token.symbol} width={75} height={75} />
+        <Text>{token.name}</Text>
+      </TokenName>
+      <TokenDetails>
+        <TokenInfo>
+          {`Staked ICX (sICX) is ICX you can move around in a liquid state and still earn staking rewards. Every sICX you hold entitles you to a share of the ICX staking pool, and the value increases over time based on the ICON reward rate.`}
+          <LineBreak />
+          {`With sICX, you can unstake faster than traditional ICX unstaking. Pay a 1% premium to swap sICX for ICX, or unstake from the wallet section to join the unstaking queue, which could take anywhere from 5 minutes, up to the regular ICX unstaking time of ~7 days.`}
```
Comment (parrot9design) [inline src/components/TokenDetails/SICXDetail.tsx:21]:
```suggestion
          {`With sICX, you can unstake faster than traditional ICX unstaking. Pay a 1% premium to swap sICX for ICX on Balanced, or unstake from the wallet section to join the unstaking queue, which could take anywhere from 5 minutes, up to the regular ICX unstaking time of ~7 days.`}
```

**Comment by BennyOptions** [comment]
Not going to work on this for now

**Pull request closed without merging**